src/components/Jobs/JobList/JobListContainer.tsx

```diff
@@ -0,0 +1,179 @@
+        props.updateDataSyncing(true)
+        const isSearchOrFilterApplied =
+            request.teams?.length || request.jobNameSearch?.length || request.appStatuses?.length
+        const updatedState = { ...state }
+        updatedState.view = JobListViewType.LOADING
+        updatedState.sortRule = {
+            key: request.sortBy,
+            order: request.sortOrder,
+        }
+        updatedState.expandedRow = {}
+        updatedState.isAllExpanded = false
+        dispatch({
+            type: JobListStateActionTypes.multipleOptions,
+            payload: updatedState,
+        })
+
+        if (abortControllerRef.current) {
+            abortControllerRef.current.abort()
+            abortControllerRef.current = null
+        }
+        abortControllerRef.current = new AbortController()
+
+        getJobs(request, { signal: abortControllerRef.current.signal })
+            .then((response) => {
+                let view = JobListViewType.LIST
+                if (response.result.jobCount === 0) {
+                    if (isSearchOrFilterApplied) {
+                        view = JobListViewType.NO_RESULT
+                    } else {
+                        view = JobListViewType.EMPTY
+                    }
+                }
+                const _jobs = jobListModal(response.result?.jobContainers)
+                dispatch({
+                    type: JobListStateActionTypes.multipleOptions,
+                    payload: {
+                        code: response.code,
+                        jobs: _jobs,
+                        isAllExpandable: _jobs.filter((job) => job.ciPipelines.length > 1).length > 0,
+                        view: view,
+                        offset: request.offset,
+                        size: response.result.jobCount,
+                        pageSize: request.size,
+                    },
+                })
+                abortControllerRef.current = null
+                props.setJobCount(response.result.jobCount)
+            })
+            .catch((errors: ServerErrors) => {
+                if (errors.code) {
+                    showError(errors)
+
+                    dispatch({
+                        type: JobListStateActionTypes.multipleOptions,
+                        payload: { code: errors.code, view: JobListViewType.ERROR },
+                    })
+                }
+            })
+            .finally(() => {
+                props.updateDataSyncing(false)
+            })
+    }
+
+    const changePage = (pageNo: number): void => {
+        let offset = state.pageSize * (pageNo - 1)
+        let qs = queryString.parse(location.search)
+        let keys = Object.keys(qs)
+        let query = {}
+        keys.map((key) => {
+            query[key] = qs[key]
+        })
+        query['offset'] = offset
+        let queryStr = queryString.stringify(query)
+        let url = `${URLS.APP}/${URLS.APP_LIST}/${URLS.APP_LIST_DEVTRON}?${queryStr}`
```

Comment:
Is it expected to go to the app list page from here?
we can directly use the URL. 
Also please avoid let wherever possible.